[3.2] docs(stepper): Revamped Stepper Overview (#833)
telerik/blazor-docs: components/stepper/overview.md

The Stepper provides several features out of the box, including the ability to define the appearance of each [step]({%slug stepper-steps-overview%}), [validation]({%slug stepper-steps-validation%}) for the steps, strict [linear flow]({%slug stepper-linear-flow%}) and [step template]({%slug step-template%}) to make the it fit any design requirements.

> ## Creating Blazor Stepper

Reviewer: The code snippet should reflect the instructions before that. Currently, the instructions do not mention anything about `Value`, `Label` and `Icon`.

The Stepper provides several features out of the box, including the ability to define the appearance of each [step]({%slug stepper-steps-overview%}), [validation]({%slug stepper-steps-validation%}) for the steps, strict [linear flow]({%slug stepper-linear-flow%}) and [step template]({%slug step-template%}) to make the it fit any design requirements.

## Creating Blazor Stepper

1. Add the `TelerikStepper` tag to add the component to your razor page.

> 2. Set the `Value` parameter. It supports one-way and two-way binding.

Reviewer: Set the Value to what? You can mention the parameter type (e.g. int). Is it zero-based or one-based?

> 2. Set the `Value` (one-way data binding) or the `bind-Value` (two-way data binding) property.

Reviewer: `bind-Value` is not exactly a property. I think `@anything` is called a directive in Blazor.

```suggestion
2. Set the `Value` parameter. It supports one-way and two-way binding.
```

> 3. Configure the [`StepperStep` instances]({%slug  stepper-steps-overview%}) inside the `StepperSteps` tag.

Reviewer: Configure what? If Label and Icon are optional, what is the minimum required configuration that we can mention here?

> 3. Under its `StepperSteps` tag include and configure the desired [`Steps`]({%slug  stepper-steps-overview%}).

Reviewer: ```suggestion
3. Configure the [`StepperStep` instances]({%slug  stepper-steps-overview%}) inside the `StepperSteps` tag.
```

}
````

## Steps

> The Steps are the building blocks of the Stepper component. You can include the desired steps by adding a `StepperStep` tag for every step. The steps appearance can be configured through the features the `StepperStep` exposes. [Read more about the Blazor Stepper configuration]({%slug stepper-steps-overview%}).

Reviewer: ```suggestion
The Steps are the building blocks of the Stepper component. Include the desired steps by adding a `StepperStep` tag for every step. Then, configure each `StepperStep` appearance via its parameters. [Read more about the Blazor Steps configuration]({%slug stepper-steps-overview%}).
```

## Orientation

> The Stepper component provides different orientations. [Read more about the Blazor Stepper orientation]({%slug stepper-orientation%}).

Reviewer: Provide some value to the sentence, not just a link.

```suggestion
The Stepper component provides horizontal and vertical orientations. [Read more about the Blazor Stepper orientation]({%slug stepper-orientation%}).
```

## Display Modes

The Stepper component provides different display modes. [Read more about the Blazor Stepper display modes]({%slug stepper-display-modes%}).

## Linear Flow

> By default, the user can select any step in the Stepper component. This behavior could be customized by configuring its Linear flow property. [Read more about the Blazor Stepper linear flow]({%slug stepper-linear-flow%}).

Reviewer: ```suggestion
By default, the user can select any step in the Stepper component and go to it directly. This behavior can be changed with the `Linear` parameter, so that the user goes through the steps sequentially. [Read more about the Blazor Stepper linear flow]({%slug stepper-linear-flow%}).
```

## Events

The Blazor Stepper generates events that you can handle and further customize its behavior. [Read more about the Blazor Stepper events]({%slug stepper-events%}).

## Parameters

The Blazor Stepper provides various parameters that allow you to configure the component:

<style>
    article style + table {
        table-layout: auto;
        word-break: normal;
    }
</style>
| Parameter | Type | Description |
| ----------- | ----------- | ----------- |
| `Value` | `int` | Defines the current step index. |
| `Orientation` | `StepperOrientation` enum <br /> `Horizontal` | Defines the orientation of the Stepper. You can read more on how to configure it in the [Orientation]({%slug stepper-orientation%}) article. |
| `Linear` | `bool` | Enables/disables [linear flow]({%slug stepper-linear-flow%}). |
| `StepType` | `StepperStepType` enum <br /> `Steps` | Defines the display mode of the Stepper. You can read more on how to configure it in the [Display modes]({%slug stepper-display-modes%}) article. |

## Next Steps

> * [Using Stepper Templates]({%slug step-template%})
> 
> * [Configure Steps]({%slug stepper-steps-overview%})
> 
> * [Explore the Stepper Events]({%slug stepper-events%})

Reviewer: The Next Steps order should make sense. First you configure steps, then you use templates. Use the same grammar form for the verbs.

```suggestion
* [Configure Steps]({%slug stepper-steps-overview%})
* [Use Stepper Templates]({%slug step-template%})
* [Explore Stepper Events]({%slug stepper-events%})
```
